main/tests.py

```diff
@@ -1 +1,73 @@
-# Create your tests here.
+import pytest
+from dogowner.models import DogOwner
+
+
+@pytest.fixture
+def create_dog_owner_user():
+    return DogOwner.create(email='dogowner@address.com',
+                           username='dogOwnerUser01',
+                           password='password123',
+                           dog_name='dog name',
+                           first_name='test',
+                           last_name='user',
+                           phone_number=1234567890,
+                           dog_race='dog race',
+                           dog_picture_url='https://www.google.com/user1.jpg',
+                           dog_age=4,
+                           dog_weight=2,
+                           dog_gender='M'
+                           )
+
+
+@pytest.mark.django_db
+class TestLoginView:
+
+    def test_enter_login_page(self, client):
+        response = client.get('/login/')
+        assert response.status_code == 200
+
+    def test_valid_login_dog_owner_user_info(self, client, create_dog_owner_user):
+        prev_logged_user = client.get('/').wsgi_request.user
+        form = {'username': 'dogOwnerUser01',
+                'password': 'password123',
+                }
+
+        response = client.post('/login/', form, follow=True)
+        curr_log_user = response.wsgi_request.user
+        assert curr_log_user == create_dog_owner_user.user
+        assert prev_logged_user != curr_log_user
+
+    def test_invalid_login_dog_owner_user_info(self, client, create_dog_owner_user):
+        forms = {'username': "daycare@address.com",
+                 'password': "incorrect",
+                 }
+        response = client.post('/login/', forms, follow=True)
+        assert create_dog_owner_user.user != response.wsgi_request.user
+        assert b"Please enter a correct username and password." in response.content
+
+
+@pytest.mark.django_db
+class TestLogoutView:
+    def test_successful_logout(self, client, create_dog_owner_user):
+        client.force_login(user=create_dog_owner_user.user)
+        logged_user = client.get('/').wsgi_request.user
+        response = client.get('/logout/')
+        assert response['Location'] == '/login/'
+        assert response.wsgi_request.user != logged_user
+
+
+@pytest.mark.django_db
+class TestIndexView:
+
+    def test_root_entrypoint_redirection_unlogged_user(self, client):
+        response = client.get("/")
+        assert response.status_code == 302
+        assert response['Location'] == '/login/'
+
+    def test_root_entrypoint_redirection_logged_user(self, client, create_dog_owner_user):
+        client.force_login(user=create_dog_owner_user.user)
+        response = client.get("/")
```

Comment:
how about testing that a logged user trying to reach /login/ will reach the homepage instead?

Reply:
Done , I added test in TestLoginView that test this case but redirect to the entry point (And from there we have test that cover redirect of logged-in user to homepage ) , because in the future we will have 2 home pages (DogOwner/DayCare) and it is looks good to me to manage this case from index view.

Reply:
I'm not sure "2 home pages" is the right approach or phrasing.

Reply:
You right . I meant 2 views for diffrent kind of users.